src/Tasks/Microsoft.NET.Build.Tasks/CreateComHost.cs

```diff
@@ -20,19 +23,89 @@ public class CreateComHost : TaskWithAssemblyResolveHooks
         [Required]
         public string ClsidMapPath { get; set; }
 
+        public ITaskItem[] TypeLibraries { get; set; }
+
         protected override void ExecuteCore()
         {
             try
             {
+                if (!TryCreateTypeLibraryIdDictionary(TypeLibraries, out var typeLibIdMap))
+                {
+                    return;
+                }
                 ComHost.Create(
                     ComHostSourcePath,
                     ComHostDestinationPath,
-                    ClsidMapPath);
+                    ClsidMapPath,
+                    typeLibIdMap);
             }
             catch (ComHostCustomizationUnsupportedOSException)
             {
                 Log.LogError(Strings.CannotEmbedClsidMapIntoComhost);
             }
+            catch (TypeLibraryDoesNotExistException ex)
+            {
+                Log.LogError(Strings.TypeLibraryDoesNotExist, ex.Path);
+            }
+            catch (InvalidTypeLibraryIdException ex)
+            {
+                Log.LogError(Strings.InvalidTypeLibraryId, ex.Id.ToString(), ex.Path);
+            }
+            catch (HResultException hr) when (hr.Win32HResult == E_INVALIDARG)
+            {
+                Log.LogError(Strings.InvalidTypeLibrary);
+            }
+        }
+
+        private bool TryCreateTypeLibraryIdDictionary(ITaskItem[] typeLibraries, out Dictionary<int, string> typeLibraryIdMap)
+        {
+            typeLibraryIdMap = null;
+            if (typeLibraries is null || typeLibraries.Length == 0)
+            {
+                return true;
+            }
+            else if (typeLibraries.Length == 1)
+            {
+                int id = 1;
+                string idMetadata = typeLibraries[0].GetMetadata("Id");
+                if (!string.IsNullOrEmpty(idMetadata))
+                {
+                    if (!int.TryParse(idMetadata, out id) || id == 0)
+                    {
+                        Log.LogError(Strings.InvalidTypeLibraryId, idMetadata, typeLibraries[0].ItemSpec);
+                        return false;
+                    }
+                }
+                typeLibraryIdMap = new Dictionary<int, string> { { id, typeLibraries[0].ItemSpec } };
+                return true;
+            }
+            typeLibraryIdMap = new Dictionary<int, string>();
+            foreach (ITaskItem typeLibrary in typeLibraries)
+            {
+                string idMetadata = typeLibrary.GetMetadata("Id");
+                if (string.IsNullOrEmpty(idMetadata))
+                {
+                    Log.LogError(Strings.MissingTypeLibraryId, typeLibrary.ItemSpec);
```

Comment:
What's the norm in the SDK for logging errors for items/collections like this? Log the first error we hit and bail out or continue going through all items and log all errors?

For example, `ResolveRuntimePackAssets` and `ResolveTargetingPackAssets` look like they all errors instead of bailing out at the first one.